plugin/evm/limit_order.go

```diff
@@ -172,60 +176,86 @@ func parseTx(orderBookABI abi.ABI, memoryDb limitorders.InMemoryDatabase, backen
 			})
 			signature := in["signature"].([]byte)
 
-			var positionType string
-			if order.BaseAssetQuantity.Int64() > 0 {
-				positionType = "long"
-			} else {
-				positionType = "short"
+			baseAssetQuantity := int(order.BaseAssetQuantity.Int64())
+			if baseAssetQuantity == 0 {
+				log.Error("order not saved because baseAssetQuantity is zero")
+				return
 			}
+			positionType := getPositionTypeBasedOnBaseAssetQuantity(baseAssetQuantity)
 			price, _ := new(big.Float).SetInt(order.Price).Float64()
-			limitOrder := limitorders.LimitOrder{
+			limitOrder := &limitorders.LimitOrder{
 				PositionType:      positionType,
 				UserAddress:       order.Trader.Hash().String(),
-				BaseAssetQuantity: int(order.BaseAssetQuantity.Uint64()),
+				BaseAssetQuantity: baseAssetQuantity,
 				Price:             price,
 				Salt:              order.Salt.String(),
+				Status:            "unfulfilled",
 				Signature:         signature,
+				BlockNumber:       blockNumber,
 				RawOrder:          in["order"],
 				RawSignature:      in["signature"],
 			}
 			memoryDb.Add(limitOrder)
 		}
-		if m.Name == "executeTestOrder" {
-			log.Info("##### in ParseTx", "executeTestOrder tx hash", tx.Hash().String())
-			go pollForReceipt(backend, tx.Hash())
-			signature := in["signature"].([]byte)
-			memoryDb.Delete(signature)
+		if m.Name == "executeMatchedOrders" {
+			signature1 := in["signature1"].([]byte)
+			memoryDb.Delete(signature1)
+			signature2 := in["signature2"].([]byte)
+			memoryDb.Delete(signature2)
 		}
 	}
 }
 
-func pollForReceipt(backend eth.EthAPIBackend, txHash common.Hash) {
-	for i := 0; i < 10; i++ {
-		receipt := getTxReceipt(backend, txHash)
-		if receipt != nil {
-			log.Info("receipt found", "tx", txHash.String(), "receipt", receipt)
-			return
-		}
-		time.Sleep(time.Second * 5)
+func callExecuteMatchedOrders(txPool *core.TxPool, orderBookABI abi.ABI, incomingOrder limitorders.LimitOrder, matchedOrder limitorders.LimitOrder) error {
+	//randomly selecting private key to get different validator profile on different nodes
+	rand.Seed(time.Now().UnixNano())
+	if rand.Intn(10000)%2 == 0 {
+		privateKey = privateKey2
```

Comment:
this way privateKey will always remain privateKey2 for the next function call too. So after the first time this condition fulfills, it will always use parivateKey2

Reply:
fixed